src/eg/src/guardian_proof.rs

```diff
@@ -0,0 +1,276 @@
+            })
+            .collect::<Vec<PossessionProofResponse>>();
+        GuardianProof {
+            i: secret_key.i,
+            c: c_vec,
+            v: v_vec,
+        }
+    }
+
+    /// This function verifies a `GuardianProof` with respect to a given `GuardianPublicKeyInfo`
+    /// This is corresponds to Verification 2 "Guardian public-key validation"
+    /// The arguments are
+    /// - self - the `GuardianProof`
+    /// - election_parameters - the election parameters
+    /// - h_p - the parameter base hash
+    /// - public_key_info - the `GuardianPublicKeyInfo`
+    pub fn validate(
+        &self,
+        election_parameters: &ElectionParameters,
+        h_p: HValue,
+        public_key_info: &dyn GuardianPublicKeyInfo,
+    ) -> Result<()> {
+        ensure!(
+            self.i == public_key_info.i(),
+            "The guardian indices of the proof and the public key must match."
+        );
+
+        let fixed_parameters = &election_parameters.fixed_parameters;
+        let varying_parameters = &election_parameters.varying_parameters;
+        let k = varying_parameters.k;
+
+        let i = self.i.get_one_based_u32();
+        let coefficient_commitments = public_key_info.coefficient_commitments();
+
+        // Equation (2.1)
+        let h_vec = (0..k.get_one_based_usize())
+            .map(|j| {
+                fixed_parameters
+                    .g
+                    .modpow(&self.v[j].0, fixed_parameters.p.borrow())
+                    * coefficient_commitments.0[j]
+                        .0
+                        .modpow(&self.c[j].0, fixed_parameters.p.borrow())
+                    % fixed_parameters.p.as_ref()
+            })
+            .collect::<Vec<BigUint>>();
+        // Verification 2 checks
+        let zero = BigUint::from(0u8);
+        let one = BigUint::from(1u8);
+        let mut verified = true;
+        for j in 0..k.get_one_based_u32() {
+            let capital_k_i_j = &coefficient_commitments.0[j as usize].0;
+            // 2.A
+            verified &= (zero <= *capital_k_i_j) && (*capital_k_i_j < *fixed_parameters.p.borrow());
+            verified &= capital_k_i_j
+                .modpow(fixed_parameters.q.borrow(), fixed_parameters.p.borrow())
+                == one;
+            // 2.B
+            verified &= (zero <= self.v[j as usize].0)
+                && (self.v[j as usize].0 < *fixed_parameters.q.borrow());
+            // 2.C
+            verified &= self.c[j as usize].0
+                == Self::challenge(h_p, i, j, capital_k_i_j, &h_vec[j as usize]).0
+        }
+        ensure!(verified, "The proof is invalid.");
+
+        Ok(())
+    }
+
+    /// Reads a `GuardianProof` from a `std::io::Read`. Validation will require the corresponding public key.
+    pub fn from_stdioread_validated(stdioread: &mut dyn std::io::Read) -> Result<Self> {
+        let self_: Self = serde_json::from_reader(stdioread).context("Reading GuardianProof")?;
+        Ok(self_)
+    }
```

Comment:
This is rather odd. Was it just a convenience method?

Reply:
Removed the function.